File: gosi-kafka-sdk-java/gosi-kafka-sdk-core/src/main/java/com/gosi/kafka/sdk/auth/SaslScramAuthHandler.java
package com.gosi.kafka.sdk.auth;

import org.apache.kafka.clients.CommonClientConfigs;
import org.apache.kafka.common.config.SaslConfigs;

import java.util.Map;

/**
 * Configures SASL/SCRAM-SHA-512 authentication.
 */
public class SaslScramAuthHandler implements AuthenticationHandler {

    private final String username;
    private final String password;
    private final boolean useTls;

    public SaslScramAuthHandler(String username, String password, boolean useTls) {
        if (username == null || username.trim().isEmpty()) {
            throw new IllegalArgumentException("Username cannot be empty");
        }
        if (password == null || password.trim().isEmpty()) {
            throw new IllegalArgumentException("Password cannot be empty");
        }
        this.username = username;
        this.password = password;
        this.useTls = useTls;
    }

    @Override
    public void configure(Map<String, Object> kafkaProperties) {
        kafkaProperties.put(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG, useTls ? "SASL_SSL" : "SASL_PLAINTEXT");
        kafkaProperties.put(SaslConfigs.SASL_MECHANISM, "SCRAM-SHA-512");
        
        String jaasTemplate = "org.apache.kafka.common.security.scram.ScramLoginModule required username=\"%s\" password=\"%s\";";
        kafkaProperties.put(SaslConfigs.SASL_JAAS_CONFIG, String.format(jaasTemplate, username, password));
    }
}

File: gosi-kafka-sdk-dotnet/Auth/SaslScramAuthHandler.cs
namespace Gosi.Kafka.Sdk.Auth;

using System;
using Confluent.Kafka;

/// <summary>
/// Configures SASL/SCRAM-SHA-512 authentication for .NET clients.
/// </summary>
public class SaslScramAuthHandler : IAuthenticationHandler
{
    private readonly string _username;
    private readonly string _password;
    private readonly bool _useTls;

    public SaslScramAuthHandler(string username, string password, bool useTls)
    {
        if (string.IsNullOrWhiteSpace(username)) throw new ArgumentException("Username cannot be empty");
        if (string.IsNullOrWhiteSpace(password)) throw new ArgumentException("Password cannot be empty");

        _username = username;
        _password = password;
        _useTls = useTls;
    }

    public void Configure(ClientConfig config)
    {
        config.SecurityProtocol = _useTls ? SecurityProtocol.SaslSsl : SecurityProtocol.SaslPlaintext;
        config.SaslMechanism = SaslMechanism.ScramSha512;
        config.SaslUsername = _username;
        config.SaslPassword = _password;
    }
}
